### PR TITLE
feat: auditlog

### LINKED ISSUE #166: Support for Audit logs
### Description

Built-in support for audit logs to track changes made in the dynamic models and core models (e.g. AppUsers, UserRole, PolicyModel, etc.)

### Background

Audit logs are essential for maintaining transparency, accountability, and compliance with regulatory requirements. They provide a detailed record of user activities, data modifications, etc. 

### Feature Details

- audit log module to capture the changes made in dynamic models
- Integration with App Panel to display the audit log table with search, filter, and export functionality

Tasks:

- [ ] Required changes like audit -> auditlog rename and removal from enterprise folder
- [ ] working version to be deployed on dev server for testing
- [ ] Code Cleaning and code review points
- [ ] check for public tenant models

### PR DESCRIPTION
**Pull request by deepakdinesh1123**
fixes #166

### CODE REVIEW

File: backend/src/zelthy/api/platform/tenancy/v1/urls.py [hunk 1]
```diff
@@ -15,6 +15,7 @@
 from zelthy.api.platform.packages.v1 import urls as packages_v1_urls
 from zelthy.api.platform.tasks.v1 import urls as tasks_v1_urls
 from zelthy.api.platform.codeassist.v1 import urls as codeassist_v1_urls
+from zelthy.api.platform.audit.v1 import urls as audit_v1_urls
```
Comment (shahharsh176) [inline backend/src/zelthy/api/platform/tenancy/v1/urls.py:18]: `audit` can be changed to `auditlog`

File: backend/src/zelthy/api/platform/tenancy/v1/urls.py [hunk 2]
```diff
@@ -57,5 +58,6 @@
     re_path(r"^(?P<app_uuid>[\w-]+)/packages/$", include(packages_v1_urls)),
     re_path(r"^(?P<app_uuid>[\w-]+)/tasks/", include(tasks_v1_urls)),
     re_path(r"^(?P<app_uuid>[\w-]+)/code-assist/", include(codeassist_v1_urls)),
+    re_path(r"^(?P<app_uuid>[\w-]+)/audit/", include(audit_v1_urls)),
```
Comment (shahharsh176) [inline backend/src/zelthy/api/platform/tenancy/v1/urls.py:61]: `auditlog`

File: backend/src/zelthy/apps/auditlog/__init__.py [hunk 3]
```diff
@@ -0,0 +1,3 @@
+# from importlib.metadata import version
+
+# __version__ = version("django-auditlog")
```
Comment (shahharsh176) [inline backend/src/zelthy/apps/auditlog/__init__.py:3]: if this is not need needed pls remove the commented code

File: backend/src/zelthy/apps/auditlog/models.py [hunk 4]
```diff
@@ -0,0 +1,605 @@
...
+    object_repr = models.TextField(verbose_name=_("object representation"))
+    serialized_data = models.JSONField(null=True)
+    action = models.PositiveSmallIntegerField(
+        choices=Action.choices, verbose_name=_("action"), db_index=True
+    )
+    changes_text = models.TextField(blank=True, verbose_name=_("change message"))
+    changes = models.JSONField(null=True, verbose_name=_("change message"))
+    tenant_actor = models.ForeignKey(
+        to="appauth.AppUserModel",
+        on_delete=models.DO_NOTHING,
```
Comment (shahharsh176) [inline backend/src/zelthy/apps/auditlog/models.py:351]: we can change the on_delete to `models.SET_NULL`

File: backend/src/zelthy/apps/auditlog/models.py [hunk 5]
```diff
@@ -0,0 +1,605 @@
...
+        blank=True,
+        null=True,
+        related_name="+",
+        verbose_name=_("tenant_actor"),
+    )
+    platform_actor = models.ForeignKey(
+        to="platformauth.PlatformUserModel",
+        on_delete=models.DO_NOTHING,
```
Comment (shahharsh176) [inline backend/src/zelthy/apps/auditlog/models.py:359]: change the on_delete to `models.SET_NULL`

File: backend/src/zelthy/apps/auditlog/models.py [hunk 6]
```diff
@@ -0,0 +1,605 @@
...
+
+            if isinstance(pk, int):
+                kwargs.setdefault("object_id", pk)
+
+            get_additional_data = getattr(instance, "get_additional_data", None)
+            if callable(get_additional_data):
+                kwargs.setdefault("additional_data", get_additional_data())
+            db = instance._state.db
+            try:
+                obj = ObjectStore.objects.get(object_uuid=instance.object_uuid)
```
Comment (shahharsh176) [inline backend/src/zelthy/apps/auditlog/models.py:60]: also handle the case where objectstore doesn't contain the object with given `instance.object_uuid`, in this case code shouldn't break, object_ref can be None in that case

File: backend/src/zelthy/apps/exportjob/models.py [hunk 7]
```diff
@@ -0,0 +1,12 @@
+from django.db import models
```
Comment (shahharsh176) [inline backend/src/zelthy/apps/exportjob/models.py:1]: please remove the exportjob as of now

File: backend/src/zelthy/config/urls_public.py [hunk 8]
```diff
@@ -20,7 +20,7 @@
     re_path(r"^", include("zelthy.apps.shared.tenancy.urls")),
 ]
 if settings.DEBUG:
-    urlpatterns += [
-        path("__debug__/", include("debug_toolbar.urls")),
-    ]
+    # urlpatterns += [
```
Comment (shahharsh176) [inline backend/src/zelthy/config/urls_public.py:23]: pls revert this change

File: backend/src/zelthy/config/urls_tenants.py [hunk 9]
```diff
@@ -7,12 +7,12 @@
     re_path(r"^", include("zelthy.apps.appauth.urls")),
     re_path(r"api/auth/", include("knox.urls")),
     re_path(r"api/", include("zelthy.api.app_auth.urls")),
-    path("__debug__/", include("debug_toolbar.urls")),
+    # path("__debug__/", include("debug_toolbar.urls")),
```
Comment (shahharsh176) [inline backend/src/zelthy/config/urls_tenants.py:10]: pls revert this change

File: backend/src/zelthy/config/urls_tenants.py [hunk 10]
```diff
@@ -7,12 +7,12 @@
...
     re_path(r"^((?:[\w\-:.,]+/)*)$", include("zelthy.apps.dynamic_models.urls")),
 ]
 
 if settings.DEBUG:
-    urlpatterns += [
-        path("__debug__/", include("debug_toolbar.urls")),
-    ]
+    # urlpatterns += [
```
Comment (shahharsh176) [inline backend/src/zelthy/config/urls_tenants.py:15]: pls revert this change

**Review by shahharsh176 — CHANGES_REQUESTED** [review]

**Comment by shahharsh176** [comment]
@deepakdinesh1123 should we change "api/platform/auditlog" folder to `api/platform/auditlogs` to be in sync with app name `auditlogs`

File: frontend/src/pages/app/components/Layout/SideMenu.jsx [hunk 11]
```diff
@@ -106,6 +106,46 @@ export default function SideMenu() {
 					Packages
 				</span>
 			</NavLink>
+			{/* <NavLink
```
Comment (kc-diabeat) [inline frontend/src/pages/app/components/Layout/SideMenu.jsx:109]: Pls remove unnecessary commented code

File: frontend/src/pages/appAccessLogs/slice/index.js [hunk 12]
```diff
@@ -0,0 +1,130 @@
+import { createSlice } from '@reduxjs/toolkit';
+
+export const appAccessLogsSlice = createSlice({
```
Comment (kc-diabeat) [inline frontend/src/pages/appAccessLogs/slice/index.js:3]: Accesslog related code should not be included in this PR.

File: backend/src/zelthy/apps/auditlogs/LICENSE [hunk 13]
```diff
@@ -0,0 +1,20 @@
+The MIT License (MIT)
```
Comment (kc-diabeat) [inline backend/src/zelthy/apps/auditlogs/LICENSE:1]: Lets have a readme file in this folder where we detail out:
- which version of auditlog library we have taken
- what changes have we made
- overall rationale
- How is it being used, how to exclude model, fields, etc.
Also include a section in Docs

**Pull request closed without merging**